auth-tokens/src/main/java/com/palantir/tokens/auth/JsonWebToken.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tokens.auth;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.guava.GuavaModule;
+import com.google.common.io.BaseEncoding;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.UUID;
+import org.immutables.value.Value;
+
+@Value.Immutable
+@Value.Style(visibility = Value.Style.ImplementationVisibility.PACKAGE, jdkOnly = true)
+public abstract class JsonWebToken {
+
+    private static final ObjectMapper MAPPER = new ObjectMapper()
+            .registerModule(new GuavaModule());
+
+    /**
+     * Returns an unverified user id.
+     */
+    @Value.Parameter
+    public abstract String getUnsafeUserId();
```

Comment:
I see. I think I'd prefer to make this a static method like
`BearerTokens#extractUnverifiedUserId(BearerToken token)` just so it's one
step further away. Thoughts?

On Wed, Jun 8, 2016 at 4:08 PM Mark Elliot <notifications@github.com> wrote:

> In auth-tokens/src/main/java/com/palantir/tokens/auth/JsonWebToken.java
> <https://github.com/palantir/auth-tokens/pull/10#discussion_r66261817>:
>
> > +import java.nio.ByteBuffer;
> > +import java.util.UUID;
> > +import org.immutables.value.Value;
> > +
> > +@Value.Immutable
> > +@Value.Style(visibility = Value.Style.ImplementationVisibility.
>
> PACKAGE, jdkOnly = true)
> > +public abstract class JsonWebToken {
> > +
> > +    private static final ObjectMapper MAPPER = new ObjectMapper()
> > +            .registerModule(new GuavaModule());
> > +
> > +    /**
> > +     * Returns an unverified user id.
> > +     */
> > +    @Value.Parameter
> > +    public abstract String getUnsafeUserId();
>
> Primarily for logging -- we'd like services to log user id without
> necessarily having to cryptographically verify (as a best effort kind of
> thing). We could forcibly add a string "-untrusted" to make it more obvious
> or mark the method as deprecated to discourage use. Some more discussion on
> #8 <https://github.com/palantir/auth-tokens/issues/8>
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/palantir/auth-tokens/pull/10/files/f99770fb97c2872150d1e8790695cb242deee7e0#r66261817>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AS4-yjdJL-4ZpR3JAWR0OmMa7Rc4AXywks5qJszsgaJpZM4IwnFn>
> .
>